### PR TITLE
docs: fix the custom agent binary file path for creating initrd image

### PR DESCRIPTION
**Pull request by YchauWang**
fix the custom agent binary file path for creating an initrd image in
the Developer-Guide.md file.

Fixes: #919

Signed-off-by: Ychau Wang <wangyongchao.bj@inspur.com>

### CODE REVIEW

File: docs/Developer-Guide.md [hunk 1]
```diff
@@ -356,7 +356,7 @@ You MUST choose one of `alpine`, `centos`, `clearlinux`, `euleros`, and `fedora`
 
 Optionally, add your custom agent binary to the rootfs with the following:
 ```
-$ sudo install -o root -g root -m 0550 -T ../../agent/kata-agent ${ROOTFS_DIR}/sbin/init
+$ sudo install -o root -g root -m 0550 -T ../../../src/agent/target/x86_64-unknown-linux-musl/release/kata-agent ${ROOTFS_DIR}/sbin/init
```
Comment (liubin) [inline docs/Developer-Guide.md:359]: Here may need take `gnu` into consideration too, like #895

Comment (YchauWang) [reply docs/Developer-Guide.md]: Thanks @liubin.

**Comment by bergwolf** [comment]
Somehow `jenkins-ci-ubuntu-1804-clh-containerd-k8s-e2e-minimal ` cannot be triggered but is marked as `required`? /cc @likebreath @GabyCT

**Comment by likebreath** [comment]
test-k8s-clh-minimal

**Comment by likebreath** [comment]
The `jenkins-ci-ubuntu-1804-clh-containerd-k8s-e2e-minimal` now is triggered. The problem is similar to what we had: the CI config has the `Excluded regions` with value of `*/.md` (which will skip for changes only on `.md` files). 

I removed this configuration for now for a quick fix, while a better solution is add a quick finish route in our CI setup where we indeed don't want to run those integration tests for changes only on `.md` file.

**Review by bergwolf — APPROVED** [review]

**Comment by jodh-intel** [comment]
For context, I believe we're talking about the `excludedRegions` XML element here:

- https://github.com/kata-containers/ci/blob/2.0-dev/jenkins/jobs/kata-containers-2.0-tests-ubuntu-PR-clh-containerd-k8s-minimal/config.xml#L83

@likebreath, @bergwolf, @GabyCT, @chavafg - fwics, we should delete the `excludedRegions` XML and rely instead on the fast return facility as that would DTRT. I've raised https://github.com/kata-containers/ci/issues/348 to discuss.

File: docs/Developer-Guide.md [hunk 2]
```diff
@@ -354,9 +354,11 @@ You MUST choose one of `alpine`, `centos`, `clearlinux`, `euleros`, and `fedora`
 >
 > - Check the [compatibility matrix](../tools/osbuilder/README.md#platform-distro-compatibility-matrix) before creating rootfs.
 
-Optionally, add your custom agent binary to the rootfs with the following:
+Optionally, add your custom agent binary to the rootfs with the following, `LIBC` default is `musl`, if `ARCH` is `ppc64le`, should set the `LIBC = gnu` and `ARCH = powerpc64le`:
 ```
-$ sudo install -o root -g root -m 0550 -T ../../agent/kata-agent ${ROOTFS_DIR}/sbin/init
+$ export ARCH = $(shell uname -m)
```
Comment (liubin) [inline docs/Developer-Guide.md:359]: Hi @YchauWang  export should have no spaces around `=`.

File: docs/Developer-Guide.md [hunk 3]
```diff
@@ -354,9 +354,11 @@ You MUST choose one of `alpine`, `centos`, `clearlinux`, `euleros`, and `fedora`
 >
 > - Check the [compatibility matrix](../tools/osbuilder/README.md#platform-distro-compatibility-matrix) before creating rootfs.
 
-Optionally, add your custom agent binary to the rootfs with the following:
+Optionally, add your custom agent binary to the rootfs with the following, `LIBC` default is `musl`, if `ARCH` is `ppc64le`, should set the `LIBC=gnu` and `ARCH=powerpc64le`:
 ```
-$ sudo install -o root -g root -m 0550 -T ../../agent/kata-agent ${ROOTFS_DIR}/sbin/init
```
Comment (liubin) [inline docs/Developer-Guide.md:359]: Thanks, from line 357, I think  the ARCH should be translated to `powerpc64le` if `unam -m` returns `ppc64le`, make the shell commands copy-and-work is helpful for users, so how about changing to:

```diff
$ export ARCH=$(shell uname -m)
- $ export LIBC=musl
$ sudo install -o root -g root -m 0550 -T ../../../src/agent/target/$(ARCH)-unknown-linux-$(LIBC)/release/kata-agent ${ROOTFS_DIR}/sbin/init

$ export ARCH=$(shell uname -m)
+ $ [ ${ARCH} == "ppc64le" ] && export LIBC=gnu || export LIBC=musl
+ $ [ ${ARCH} == "ppc64le" ] && export ARCH=powerpc64le
$ sudo install -o root -g root -m 0550 -T ../../../src/agent/target/$(ARCH)-unknown-linux-$(LIBC)/release/kata-agent ${ROOTFS_DIR}/sbin/init
```

Comment (YchauWang) [reply docs/Developer-Guide.md:359]: Thanks @liubin, yes, you are right.

**Review by liubin — APPROVED** [review]

**Pull request merged**